src/cpp/session/modules/SessionProfiler.cpp

```diff
@@ -65,10 +65,40 @@ void handleProfilerResReq(const http::Request& request,
    pResponse->setCacheableFile(profileResource, request);
 }
 
+void onDocPendingRemove(const std::string &id)
+{
+   // retrieve document from source database
+   boost::shared_ptr<source_database::SourceDocument> pDoc(
+               new source_database::SourceDocument());
+   Error error = source_database::get(id, pDoc);
```

Comment:
Changed, this makes a lot of sense especially if we keep adding subscribers to this event.